File: MediaData/src/main/java/com/xxx/mediadata/utils/SortOrder.java
package com.xxx.mediadata.utils;

import android.provider.MediaStore;

/**
 * desc:
 * *
 * user: xujj
 * time: 2022/8/26 17:18
 **/
public class SortOrder {

    public final static String DATE_ADDED_DESC = MediaStore.Audio.Media.DATE_ADDED + " desc";
    public final static String TITLE_A_Z = MediaStore.Audio.Media.TITLE;
    public final static String TITLE_Z_A = MediaStore.Audio.Media.TITLE + " desc";
    public final static String DURATION = MediaStore.Audio.Media.DURATION;
    public final static String DURATION_DESC = MediaStore.Audio.Media.DURATION + " desc";
}

File: MediaData/src/main/java/com/tool/mediadata/utils/SortOrder.kt
package com.tool.mediadata.utils

import android.provider.MediaStore

/**
 * desc:
 **
 * user: xujj
 * time: 2024/8/15 18:15
 **/

object SortOrder {
    const val _ID = MediaStore.Audio.Media._ID
    const val DATE_ADDED = MediaStore.Audio.Media.DATE_ADDED
    const val DATE_ADDED_DESC = MediaStore.Audio.Media.DATE_ADDED + " desc"
    const val TITLE_A_Z = MediaStore.Audio.Media.TITLE
    const val TITLE_Z_A = MediaStore.Audio.Media.TITLE + " desc"
    const val DURATION = MediaStore.Audio.Media.DURATION
    const val DURATION_DESC = MediaStore.Audio.Media.DURATION + " desc"

    object ArtistSortOrder {
        const val ARTIST_A_Z = "ARTIST_A_Z"
        const val ARTIST_Z_A = "ARTIST_Z_A"

        const val ARTIST_MUSIC_COUNT = "ARTIST_MUSIC_COUNT"
        const val ARTIST_MUSIC_COUNT_DESC = "ARTIST_MUSIC_COUNT_DESC"
    }

    object AlbumSortOrder {
        const val ALBUM_A_Z = "ALBUM_A_Z"
        const val ALBUM_Z_A = "ALBUM_Z_A"

        const val ALBUM_MUSIC_COUNT = "ALBUM_MUSIC_COUNT"
        const val ALBUM_MUSIC_COUNT_DESC = "ALBUM_MUSIC_COUNT_DESC"
    }

    object FolderSortOrder {
        const val FOLDER_A_Z = "FOLDER_A_Z"
        const val FOLDER_Z_A = "FOLDER_Z_A"

        const val FOLDER_MUSIC_COUNT = "FOLDER_MUSIC_COUNT"
        const val FOLDER_MUSIC_COUNT_DESC = "FOLDER_MUSIC_COUNT_DESC"
    }
}
